include/seqan3/alignment/pairwise/execution/execution_handler_parallel.hpp

```diff
@@ -0,0 +1,169 @@
+
+/*!\file
+ * \brief Provides seqan3::detail::execution_handler_parallel.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <functional>
+#include <thread>
+#include <type_traits>
+#include <vector>
+
+#include <seqan3/contrib/parallel/buffer_queue.hpp>
+#include <seqan3/core/parallel/detail/reader_writer_manager.hpp>
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/concepts>
+
+#include <seqan3/core/debug_stream.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Handles the parallel execution of alignments.
+ * \ingroup execution
+ *
+ * \details
+ *
+ * At the moment this class cannot be reused for multiple calls. Thus the executing buffer must call all
+ * alignments at once and not in multiple iterations.
+ */
+class execution_handler_parallel
+{
+private:
+
+    //!\brief The task type.
+    using task_type = std::function<void()>;
+
+public:
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+
+    /*!\brief Constructs the execution handler with given thread count.
+     * \param thread_count The thread count to use.
+     *
+     * \details
+     *
+     * Starts `thread_count` many threads processing the tasks in the queue in parallel.
+     */
+    execution_handler_parallel(size_t const thread_count) : state{new internal_state{}}
+    {
+        auto * q = &(state->queue);
+        for (size_t i = 0; i < thread_count; ++i)
+        {
+            state->thread_pool.emplace_back([q] ()
+            {
+                for (;;)
+                {
+                    task_type task;
+                    if (q->wait_pop(task) == contrib::queue_op_status::closed)
+                        return;
+
+                    task();
+                }
+            });
+        }
+    }
+
+    //!\brief Constructs the execution handler with std::thread::hardware_concurrency.
+    execution_handler_parallel() : execution_handler_parallel{std::thread::hardware_concurrency()}
+    {}
+
+    execution_handler_parallel(execution_handler_parallel const &) = delete;                 //!< Deleted.
+    execution_handler_parallel(execution_handler_parallel &&) = default;                     //!< Defaulted.
+    execution_handler_parallel & operator=(execution_handler_parallel const &) = delete;     //!< Deleted.
+    execution_handler_parallel & operator=(execution_handler_parallel &&) = default;         //!< Defaulted.
+
+    //!\brief Waits for threads to finish.
+    ~execution_handler_parallel()
+    {
+        if (state != nullptr)
+            wait();
+    }
+    //!\}
+
+    /*!\name Execution
+     * \{
+     */
+    /*!\brief Invokes the passed alignment instance in a non-blocking manner.
+     * \copydetails execution_handler_sequential::execute
+     */
+
+     // * \tparam fn_type           The callable that needs to be invoked; must model std::Invocable with first_range_type
+     // *                           and second_range_type.
```

Comment:
what happened to the documentation here?

Reply:
yes. I wanted to check this if this works well in doxygen. Haven't tested it yet.

Reply:
And now you just deleted it?